CRM/Certificate/Service/Certificate.php

```diff
@@ -25,12 +25,13 @@ public function store($values) {
       if (!empty($values['id'])) {
         $params['id'] = $values['id'];
       }
+      $isMembershipCertificate = (int) $values['type'] === CRM_Certificate_Enum_CertificateType::MEMBERSHIPS;
       $params['name'] = $values['name'];
       $params['entity'] = $values['type'];
       $params['end_date'] = $values['end_date'];
       $params['start_date'] = $values['start_date'];
-      $params['min_valid_from_date'] = $values['min_valid_from_date'] ?? NULL;
-      $params['max_valid_through_date'] = $values['max_valid_through_date'] ?? NULL;
+      $params['min_valid_from_date'] = isset($values['min_valid_from_date']) && $isMembershipCertificate ? $values['min_valid_from_date'] : NULL;
```

Comment:
@shahrukh-compuco look at how we store extra values for entities here

Example of the event certificate here
https://github.com/compucorp/uk.co.compucorp.certificate/blob/7d66548ebf5e19b6c5f0c477eac90de569063f8f/CRM/Certificate/Service/CertificateEvent.php#L37

For membership certificate, the extra values method will be implemented here

https://github.com/compucorp/uk.co.compucorp.certificate/blob/7d66548ebf5e19b6c5f0c477eac90de569063f8f/CRM/Certificate/Service/CertificateMembership.php

Reply:
These are not extra values instead these are certificate properties but its just that they depend on the certificate type to be membership and in storeExtraValues we dont save certificate properties as its already saved before calling the storeExtraValues